primitives/src/adapter.rs

```diff
@@ -59,7 +59,7 @@ pub struct Session {
     pub uid: ValidatorId,
 }
 
-pub trait Adapter: ChannelValidator + Send + Clone + Debug {
+pub trait Adapter: ChannelValidator + Send + Sync + Clone + Debug {
```

Comment:
Do we need `Sync` or should we just clone the `Adapter` instead of passing a reference to it?